dotcom-rendering/src/web/components/ServerSideTestProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import type { PartialStoryFn } from '@storybook/addons/dist/ts3.9/types';
+import type { ComponentProps, JSXElementConstructor } from 'react';
+import React from 'react';
+import type { ServerSideTests } from '../../types/config';
+
+const ServerSideTestContext = React.createContext<
+	{ activeServerSideTests: ServerSideTests } | undefined
+>(undefined);
+
+export const ServerSideTestProvider = ({
+	tests,
+	children,
+}: {
+	tests: ServerSideTests;
+	children: React.ReactNode;
+}) => (
+	<ServerSideTestContext.Provider value={{ activeServerSideTests: tests }}>
+		{children}
+	</ServerSideTestContext.Provider>
+);
+
+/**
+ * Get current server side tests
+ *
+ * @example
+ * const { activeServerSideTests } = useServerSideTests();
+ *
+ * if (activeServerSideTests.myTestName == "variant") {
+ * 	//...
+ * } else {
+ * 	//...
+ * }
+ *
+ */
+export const useServerSideTests = () => {
+	const context = React.useContext(ServerSideTestContext);
+
+	if (context === undefined) {
+		throw Error(
+			'useServerSideTests must be used within the ContentABTestProvider',
+		);
+	}
+
+	return context;
+};
+
+/**
```

Comment:
I've adapted the wrapper component and the hook code above from `ContentABTest.tsx` in AMP (which in turn follows [the pattern recommended by Kent C. Dodds](https://kentcdodds.com/blog/how-to-use-react-context-effectively).

The Storybook decorator is something new that I've come up with today (which probably shows when you read the code!) It feels like Storybook/Chromatic might be one of the main pain points for using Context in this way, so I thought it would be worth trying to create a user-friendly way to mock the `ServerSideTestsProvider` in Storybook.

Using a factory here doesn't feel ideal tbh. I ended up doing it to make the type declarations easier (for me, at least!) but I'm sure that it could be rewritten in a more user-friendly way if the pattern is adopter.